tests/tools/fswalker/reporter/reporter_test.go

```diff
@@ -0,0 +1,134 @@
+package reporter
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+
+	"github.com/golang/protobuf/ptypes/timestamp"
+	fspb "github.com/google/fswalker/proto/fswalker"
+
+	"github.com/kopia/kopia/tests/testenv"
+)
+
+func TestReporterWithFiles(t *testing.T) {
+	ctx := context.TODO()
+
+	config := &fspb.ReportConfig{
+		Version:    1,
+		ExcludePfx: nil,
+	}
+
+	fileList := []*fspb.File{
+		{
+			Version: 0,
+			Path:    filepath.Join("some", "path"),
+			Info: &fspb.FileInfo{
+				Name: "this_is_a.file",
+				Size: 11235,
+				Mode: 0700,
+				Modified: &timestamp.Timestamp{
+					Seconds: 12,
+					Nanos:   0,
+				},
+				IsDir: false,
+			},
+			Stat: &fspb.FileStat{
+				Dev:     0,
+				Inode:   0,
+				Nlink:   0,
+				Mode:    0,
+				Uid:     0,
+				Gid:     0,
+				Rdev:    0,
+				Size:    0,
+				Blksize: 0,
+				Blocks:  0,
+				Atime: &timestamp.Timestamp{
+					Seconds: 0,
+					Nanos:   0,
+				},
+				Mtime: &timestamp.Timestamp{
+					Seconds: 0,
+					Nanos:   0,
+				},
+				Ctime: &timestamp.Timestamp{
+					Seconds: 0,
+					Nanos:   0,
+				},
+			},
+			Fingerprint: nil,
```

Comment:
In the actual data consistency checks we want to leverage the Fingerprint, which by default is SHA256